### PR TITLE
core: remove dangling interfaces from host in case of error

### PR DESCRIPTION
**Pull request by flouthoc**
When invoking `setup` with same config twice we get an error that interface
already exists which is expected. But even then we leave two dandling interface
on host.

* First the one we created for host.
* Second which we were unable to move to container namespace.

We must clean dangling interfaces in case of error. So teardown is not a
mess.

This solves the issue when two consecutive setup commands for same configurations
is invoked.

**How to reproduces the problem solved which is being solved in this PR ?**

```console
sudo ./bin/netavark -f src/test/config/setupopts2.test.json setup /var/run/netns/netnstest
sudo ./bin/netavark -f src/test/config/setupopts2.test.json setup /var/run/netns/netnstest
```

Second setup will fail as it is expected but would also leave dangling
veth pairs on host. Which will make future teardown difficult.

### CODE REVIEW

**Comment by flouthoc** [comment]
@Luap99 @mheon @baude @lsm5 This should fix the issues which we are facing with dangling interfaces.

**Comment by flouthoc** [comment]
Once this merged and all leaking interfaces are cleaned up from host manually. There should be no issues ever of any dangling interfaces.

File: src/network/core.rs [hunk 1]
```diff
@@ -119,10 +119,15 @@ impl Core {
         ) {
             Ok(_) => (),
             Err(err) => {
+                // it seems something went wrong
+                // we must not leave dangling interfaces
+                // otherwise cleanup would become mess
+                // try removing leaking interfaces from host
+                let _ = core_utils::CoreUtils::remove_interface(&host_veth_name);
```
Comment (Luap99) [inline src/network/core.rs:126]: Can you log the error if there is one, maybe as warning.

Comment (flouthoc) [reply src/network/core.rs]: sure

File: src/network/core.rs [hunk 2]
```diff
@@ -119,10 +120,17 @@ impl Core {
         ) {
             Ok(_) => (),
             Err(err) => {
+                // it seems something went wrong
+                // we must not leave dangling interfaces
+                // otherwise cleanup would become mess
+                // try removing leaking interfaces from host
+                if let Err(er) = core_utils::CoreUtils::remove_interface(&host_veth_name) {
```
Comment (baude) [inline src/network/core.rs:127]: this is sort of a blind remove? should we be logging that?

Comment (flouthoc) [reply src/network/core.rs:127]: Its a blind remove cause we created a veth pair on host but whole call failed so we must remove it.
We are loggin if we failed to remove this.

Comment (flouthoc) [reply src/network/core.rs:127]: Only problem is when we are unable to remove this and we are warning when we fail to do it.

**Pull request merged**